rust/lesson-plan.md

```diff
@@ -0,0 +1,315 @@
+
+See [README.md] for the overview, goals, audience, and prerequisites.
+
+- [Prerequisites](#user-content-prerequisites)
+- [Getting the materials](#user-content-getting-the-materials)
+- [Course structure](#user-content-course-structure)
+- [Getting help](#user-content-getting-help)
+- [Making PNA Rust Better](#user-content-making-pna-rust-better)
+- [Practical Networked Applications in Rust](#user-content-practical-networked-applications-in-rust)
+  - [Building Blocks 1](#user-content-building-blocks-1)
+  - [Project 1: The Rust toolbox and strong foundations](#user-content-project-1-the-rust-toolbox-and-strong-foundations)
+  - [Building Blocks 2](#user-content-building-blocks-2)
+  - [Project 2: Log-structured file I/O](#user-content-project-2-log-structured-file-io)
+  - [Building Blocks 3](#user-content-building-blocks-3)
+  - [Project 3: Synchronous client-server networking](#user-content-project-3-synchronous-client-server-networking)
+  - [Building Blocks 4](#user-content-building-blocks-4)
+  - [Project 4: Parallelism](#user-content-project-4-parallelism)
+  - [Building Blocks 5](#user-content-building-blocks-5)
+  - [Project 5: Asynchronous programming in Rust](#user-content-project-5-asynchronous-programming-in-rust)
+
+
+## Prerequisites
+
+As [described in the README.md][pre], this is not a course for novice
+programmers, and there are significant prerequisites. Ensure that you meet them
+all before proceeding.
+
+
+## Getting the materials
+
+All material for this course is in the
+
+> https://github.com/pingcap/talent-training
+
+git repository on GitHub, in the [`rust` subdirectory][rs]. You may want a copy
+of it on your local computer, particularly for easy access to the conformance
+tests for each project.
+
+Each project builds on experience from the previous project. It is reasonable to
+start each project by copying the previous.
+
+
+## Course structure
+
+The overall arc of the course is defined by a series of projects that
+incrementally introduce new subjects important to systems programming in Rust.
+
+Because building an entire database from scratch while also learning all the
+concepts involved is a daunting task, each project is proceeded by a "building
+blocks" section, where individual concepts are explored individually. These
+building blocks will consist of external readings, external programming
+problems, and other single-subject content.
+
+The building blocks sections are an opportunity to clear your mind, step away
+from the larger project, and focus your attention on learning and practicing a
+single subject in isolation. **Do not pass them up**.
+
+Each project builds on the last &mdash; the APIs, command-line interfaces, and
+parts of the implementation will often remain the same between projects, while
+you only integrate new features and improvements related to a project's subject
+matter. As such, it is ok to begin each project by copying your source code from
```

Comment:
I think I'll change it to say that both are reasonable. There are a few reasons I think it could be desirable to keep the individual projects, though a git tag would also work fine.

Also, this paragraph is in the wrong section! Moving it.

Reply:
Actually, I'll say "start where you left off (but you may want to add a [git tag])" or something.

Reply:
I'll also update all the project setup texts to say continue where you left off.